# Exploratory Data Analysis

In [18]:
import pandas as pd 
import numpy as np
import matplotlib.pyplot as plt

In [19]:
df = pd.read_csv("abalonedata.csv", header=None, names=["sex", "length", "diameter", "height", "weight", "shucked_weight", "viscera_weight", "shell_weight", "rings"])

In [20]:
df.shape

(4177, 9)

In [21]:
df.head()

sex  length  diameter  height  weight  shucked_weight  viscera_weight  \
0   M   0.455     0.365   0.095  0.5140          0.2245          0.1010   
1   M   0.350     0.265   0.090  0.2255          0.0995          0.0485   
2   F   0.530     0.420   0.135  0.6770          0.2565          0.1415   
3   M   0.440     0.365   0.125  0.5160          0.2155          0.1140   
4   I   0.330     0.255   0.080  0.2050          0.0895          0.0395   

   shell_weight  rings  
0         0.150     15  
1         0.070      7  
2         0.210      9  
3         0.155     10  
4         0.055      7

In [22]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4177 entries, 0 to 4176
Data columns (total 9 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   sex             4177 non-null   object 
 1   length          4177 non-null   float64
 2   diameter        4177 non-null   float64
 3   height          4177 non-null   float64
 4   weight          4177 non-null   float64
 5   shucked_weight  4177 non-null   float64
 6   viscera_weight  4177 non-null   float64
 7   shell_weight    4177 non-null   float64
 8   rings           4177 non-null   int64  
dtypes: float64(7), int64(1), object(1)
memory usage: 293.8+ KB


In [23]:
df.describe()

length     diameter       height       weight  shucked_weight  \
count  4177.000000  4177.000000  4177.000000  4177.000000     4177.000000   
mean      0.523992     0.407881     0.139516     0.828742        0.359367   
std       0.120093     0.099240     0.041827     0.490389        0.221963   
min       0.075000     0.055000     0.000000     0.002000        0.001000   
25%       0.450000     0.350000     0.115000     0.441500        0.186000   
50%       0.545000     0.425000     0.140000     0.799500        0.336000   
75%       0.615000     0.480000     0.165000     1.153000        0.502000   
max       0.815000     0.650000     1.130000     2.825500        1.488000   

       viscera_weight  shell_weight        rings  
count     4177.000000   4177.000000  4177.000000  
mean         0.180594      0.238831     9.933684  
std          0.109614      0.139203     3.224169  
min          0.000500      0.001500     1.000000  
25%          0.093500      0.130000     8.000000  
50%          0.171000      0.234000     9.000000  
75%          0.253000      0.329000    11.000000  
max          0.760000      1.005000    29.000000

In [24]:
df["sex"].value_counts()

M    1528
I    1342
F    1307
Name: sex, dtype: int64

In [25]:
df.duplicated(keep='first').sum()

0

In [27]:
print(df.where(df["height"] == 0.0))

      sex  length  diameter  height  weight  shucked_weight  viscera_weight  \
0     NaN     NaN       NaN     NaN     NaN             NaN             NaN   
1     NaN     NaN       NaN     NaN     NaN             NaN             NaN   
2     NaN     NaN       NaN     NaN     NaN             NaN             NaN   
3     NaN     NaN       NaN     NaN     NaN             NaN             NaN   
4     NaN     NaN       NaN     NaN     NaN             NaN             NaN   
...   ...     ...       ...     ...     ...             ...             ...   
4172  NaN     NaN       NaN     NaN     NaN             NaN             NaN   
4173  NaN     NaN       NaN     NaN     NaN             NaN             NaN   
4174  NaN     NaN       NaN     NaN     NaN             NaN             NaN   
4175  NaN     NaN       NaN     NaN     NaN             NaN             NaN   
4176  NaN     NaN       NaN     NaN     NaN             NaN             NaN   

      shell_weight  rings  
0              NaN    N

- There are 4177 rows in the dataframe.
- The sex column is categorised into 3 options (Nominal data), but the other 8 columns are numerical. The rings column is discrete data whilst the other columns are continuous data.
- There are no null values in the data.
- No rows are duplicated.
